### PR TITLE
Refactor bat-native-ads for migration to brave-core

### LINKED ISSUE #76: Refactor bat-native-ads for migration to brave-core

### PR DESCRIPTION
**Pull request by tmancey**
fixes https://github.com/brave-intl/bat-native-ads/issues/76
requires https://github.com/brave/brave-core/pull/1806

### CODE REVIEW

File: src/client.cc [hunk 1]
```diff
@@ -106,7 +111,9 @@ void Client::FlagShoppingState(
   client_state_->shop_url = url;
   client_state_->score = score;
 
-  client_state_->last_shop_time = helper::Time::Now();
+  auto now = base::Time::Now();
```
Comment (emerick) [inline src/client.cc]: Seems like `NowInSeconds` would be a good candidate for a member function, since we're doing it so much.

**Review by emerick — APPROVED** [review]
LGTM, one minor suggestion

File: src/client.cc (hunk 1 above)
Comment (tmancey) [reply src/client.cc]: 👍 good spot, fixed

**Review by emerick — APPROVED** [review]
LGTM

**Review by NejcZdovc — APPROVED** [review]
re-approving

**Pull request merged**